[esx_addons]/esx_lscustom/client/main.lua

```diff
@@ -1,542 +1,434 @@
-                            if IsControlJustReleased(0, 38) then
-                                lsMenuIsShowed = true
-
-                                local vehicle = GetVehiclePedIsIn(playerPed, false)
-                                FreezeEntityPosition(vehicle, true)
-                                myCar = ESX.Game.GetVehicleProperties(vehicle)
-
-                                ESX.UI.Menu.CloseAll()
-                                GetAction({
-                                    value = 'main'
-                                })
-
-                                -- Prevent Free Tunning Bug
-                                CreateThread(function()
-                                    while true do
-                                        local Sleep = 1000
-                                        if lsMenuIsShowed then
-                                            Sleep = 0
-                                            DisableControlAction(2, 288, true)
-                                            DisableControlAction(2, 289, true)
-                                            DisableControlAction(2, 170, true)
-                                            DisableControlAction(2, 167, true)
-                                            DisableControlAction(2, 168, true)
-                                            DisableControlAction(2, 23, true)
-                                            DisableControlAction(0, 75, true) -- Disable exit vehicle
-                                            DisableControlAction(27, 75, true) -- Disable exit vehicle
-                                        end
-                                        Wait(0)
-                                    end
-                                end)
-                            end
-                        end
-                    end
-                end
-                if not Near and HintDisplayed then
-                    HintDisplayed = false
-                    ESX.HideUI()
-                end
-            end
-        end
-        Wait(Sleep)
-    end
+	while true do
+	local Sleep = 1500
+		local playerPed = PlayerPedId()
+
+		if IsPedInAnyVehicle(playerPed, false) then
+			local coords = GetEntityCoords(playerPed)
+			local currentZone, zone, lastZone
+
+			if (ESX.PlayerData.job and ESX.PlayerData.job.name == 'mechanic') or not Config.IsMechanicJobOnly then
+				for k,v in pairs(Config.Zones) do
+					local zonePos = vector3(v.Pos.x, v.Pos.y, v.Pos.z)
+					if #(coords - zonePos) < v.Size.x then 
+						Sleep = 0
+						if not lsMenuIsShowed then
+							ESX.ShowHelpNotification(v.Hint)
+							if IsControlJustReleased(0, 38) then
+								lsMenuIsShowed = true
+
+								local vehicle = GetVehiclePedIsIn(playerPed, false)
+								FreezeEntityPosition(vehicle, true)
+								myCar = ESX.Game.GetVehicleProperties(vehicle)
+								TriggerServerEvent('esx_lscustom:changeVehicleData', "add", {plate = GetVehicleNumberPlateText(vehicle), props = myCar})
```

Comment:
You can replace this line with :

```lua
TriggerServerEvent('esx_lscustom:changeVehicleData', "add", myCar)
```

Reply:
Some people use space plate in esx_vehicleshop, in code we need format that to send it to server side
```
Config.PlateLetters  = 3
Config.PlateNumbers  = 3
Config.PlateUseSpace = true
```